docker/build.bzl

```diff
@@ -11,7 +11,35 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Rule for building a Docker image."""
+"""Rule for building a Docker image.
+
+In addition to the base docker_build rule, we expose its constituents
+(attr, outputs, implementation) directly so that others may expose a
+more specialized build leveraging the same implementation.  The
+expectation in such cases is that users will write something like:
+
+  load(
+    "@io_bazel_rules_docker//docker:build.bzl",
+    _build_attrs="attrs",
```

Comment:
That would be good to respect the naming we do native sandwich.

E.g. that would be:
_docker="docker"

docker would then be a struct inside build.bzl:
docker = struct(
  build = struct(
      impl=_implementation,
      attrs=_attrs,
      outputs=_outputs
  )
)

And you can call _docker.build.impl directly. Also we might want to switch to [named provider](https://bazel.build/designs/skylark/declared-providers.html) if we start offering interaction API